apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/supervisor/client/impl/MessageDispatcherImpl.java

```diff
@@ -103,17 +120,16 @@ void addConnection(URI uri, boolean isDefault) {
         client.connect();
     }
 
-    private SvBusResponse getResponse(Long rqId) throws SocketTimeoutException {
-        SvBusResponse res = null;
+    private <T extends SvBusResponse> T getResponse(Long rqId) throws SocketTimeoutException {
+        T res = null;
         ResponseLatch l = waiting.get(rqId);
         if (l != null) {
             res = l.get(RESPONSE_WAIT_TIMEOUT_MS);
         }
         return res;
     }
 
-    public SvBusResponse sendSync(SvBusMessage rq, String path, URI addr) {
-        SvBusResponse res = null;
+    public <T extends SvBusResponse> T sendSync(SvBusMessage rq, String path, URI addr) throws MessageSendingException, SocketTimeoutException {
```

Comment:
Do we really need to ask a user of this mehod to catch throws MessageSendingException, SocketTimeoutException?
Might be better to deal with them inside a method and just throw?

Reply:
There is nothing we can do with those exceptions and such code force the user to handle it to not cause unexpected system fail

Reply:
why not just throw this exception inside this method so that not to ask a user to do so?

Reply:
why not just throw this exception inside this method so that not to ask a user to do so?

Reply:
Fixed